src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -0,0 +1,71 @@
+__all__ = ("create_name_mapping_layers", "get_mri", "get_reverse_mri")
+
+
+from enum import Enum
+from typing import Dict, Union, cast
+
+from sentry.snuba.metrics.naming_layer.mri import SessionMRI, TransactionMRI
+from sentry.snuba.metrics.naming_layer.public import SessionMetricKey, TransactionMetricKey
+from sentry.snuba.sessions_v2 import InvalidField
+
+
+def create_name_mapping_layers() -> None:
+    # ToDo(ahmed): Hack this out once the FE changes their mappings
+    # Backwards Compat
+    NAMING_MAPPING_LAYER.update(
+        {
+            # Session
+            "sentry.sessions.session": SessionMRI.SESSION,
+            "sentry.sessions.user": SessionMRI.USER,
+            "sentry.sessions.session.duration": SessionMRI.RAW_DURATION,
+            "sentry.sessions.session.error": SessionMRI.ERROR,
+        }
+    )
+
+    for (MerticKey, MRI) in (
+        (SessionMetricKey, SessionMRI),
+        (TransactionMetricKey, TransactionMRI),
+    ):
+        # Adds new names at the end, so that when the reverse mapping is created
+        for metric_key in MerticKey:
+            try:
+                NAMING_MAPPING_LAYER[metric_key.value] = MRI[metric_key.name]
+            except KeyError:
+                continue
```

Comment:
> Why would this ever throw a KeyError? Wouldn't that mean that we failed to configure an MRI for metric key? In that case I would just let the exception bubble up instead of silently catching it.

Agreed

> Maybe it would even make sense to define metric keys and MRIs all in one place, something like

If I don't use Enums , I will end up having to write the string everywhere else in the code, and I'd rather steer away from that

Reply:
You could use these `MetricIdentifier` objects everywhere instead of the enum objects, and then have `MetricIdentifier.from_mri()` and `MetricIdentifier.from_name()` methods to get them... but this is just an idea I haven't fully thought through, so feel free to ignore.